core/templates/components/ck-editor-helpers/ck-editor-4-rte.component.ts

```diff
@@ -275,7 +276,13 @@ export class CkEditor4RteComponent implements AfterViewInit, OnChanges,
         this.contextService.isExplorationLinkedToStory() &&
               AppConstants.VALID_RTE_COMPONENTS_FOR_ANDROID.indexOf(
                 componentDefn.id) === -1);
-      if (!(hideComplexExtensionFlag || notSupportedOnAndroidFlag)) {
+      var InvalidForBlogPostEditor = (
+        this.contextService.isInBlogPostEditorPage() &&
+        AppConstants.INVALID_RTE_COMPONENTS_FOR_BLOG_POST_EDITOR.indexOf(
+          componentDefn.id) !== -1);
```

Comment:
Make a separate function for this condition with a brief explanation of what it does.

Reply:
Done